src/com/lld/inmemorycache/model/impl/LRUEvictionPolicy.java

```diff
@@ -0,0 +1,70 @@
+package com.lld.inmemorycache.model.impl;
+
+import com.lld.inmemorycache.model.DoublyLinkedList;
+import com.lld.inmemorycache.model.IEvictionPolicy;
+import com.lld.inmemorycache.model.Node;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class LRUEvictionPolicy implements IEvictionPolicy {
+    private DoublyLinkedList keys;
+    private HashMap<String, Node> mapper;
+
+    private ReentrantLock lock;
+
+    public LRUEvictionPolicy() {
+        keys = new DoublyLinkedList();
+        mapper = new LinkedHashMap<>();

```

Comment:
Looks like managing a hash map here just to find out if the eviction policy has a key or not is an overkill. Already storage layer is maintaining another hash map. If we maintain a map here, it just doubles the space with not much gain. Probably we can simply remove / get the key from the double linked list when a key is accessed or removed from the cache and return either the node or `null` to differentiate whether the operation was successful.